src/Products/Annotation/Controllers/AnnotationApiController.cs

```diff
@@ -592,145 +427,38 @@ public HttpResponseMessage Annotate(AnnotationPostedDataEntity annotateDocumentR
-            try
-            {
-
-                Stream document = File.Open(path, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
-                List<PageImage> pageImages = AnnotationImageHandler.GetPages(document, new ImageOptions());
-                document.Dispose();
-                document.Close();
-                for (int i = 0; i < pageImages.Count; i++)
-                {
-                    PageDataDescriptionEntity page = new PageDataDescriptionEntity();
-                    if (pageImages != null)
-                    {
-                        byte[] bytes;
-                        using (MemoryStream memoryStream = new MemoryStream())
-                        {
-                            Stream imageStream = pageImages[i].Stream;
-                            imageStream.Position = 0;
-                            imageStream.CopyTo(memoryStream);
-                            bytes = memoryStream.ToArray();
-                        }
-                        string encodedImage = Convert.ToBase64String(bytes);
-                        page.SetData(encodedImage);
-                    }
-                    description.pages.Add(page);
-                }
-
-                return description.pages;
-            }
-            catch (FileNotFoundException ex)
-            {
-                throw ex;
-            }
-        }
+        public void RemoveAnnotations(string documentGuid, string password) {
+            // TODO: consider global properties
+            string tempFilename = Path.GetFileNameWithoutExtension(documentGuid) + "_tmp";
+            string tempPath = Path.Combine(Path.GetDirectoryName(documentGuid), tempFilename + Path.GetExtension(documentGuid));
 
-        public void RemoveAnnotations(string path)
-        {
             try
             {
-                Stream resultStream = null;
-                string tempFilePath = "";
-                using (Stream inputStream = File.Open(path, FileMode.Open, FileAccess.ReadWrite, FileShare.ReadWrite))
+                using (Stream inputStream = File.Open(documentGuid, FileMode.Open, FileAccess.ReadWrite, FileShare.ReadWrite))
                 {
-                    resultStream = AnnotationImageHandler.RemoveAnnotationStream(inputStream);
-                    resultStream.Position = 0;
-                    tempFilePath = Resources.GetFreeFileName(GlobalConfiguration.Annotation.GetFilesDirectory(), Path.GetFileName(path));
-                    using (Stream tempFile = File.Create(tempFilePath))
+                    using (GroupDocs.Annotation.Annotator annotator = new GroupDocs.Annotation.Annotator(inputStream, GetLoadOptions(password, true)))
                     {
-                        resultStream.Seek(0, SeekOrigin.Begin);
-                        resultStream.CopyTo(tempFile);
+                        annotator.Save(tempPath, new SaveOptions() { AnnotationTypes = AnnotationType.None });
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove these redundant parentheses.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Annotation-for-.NET-MVC/pullRequest?prid=5372470)